### ISSUE TITLE
Allow checkbox's TextStyle modification

### ISSUE THREAD

**Issue by Mubelotix** [emilk/egui#406]
**Feature request**
Make it possible to change the `TextStyle` of checkboxes, so that there is a basic font size support before #56.

**Solution**
```rust
Checkbox::new(&mut enabled, "Enable").text_style(TextStyle::Body);
```

**Alternatives**
Change the `TextStyle::Button` font size, but that would break other buttons.

**Comment by emilk** [comment]
There are many widgets we want to change text styles and text color on. Adding modifiers for them all may be the wrong approach.

An alternative is to instead set a new default in `Style`. This would also make it more convenient to change it for a lot of checkboxes:

``` rust
ui.style().override_text_style = Some(TextStyle::Monospace);
ui.checkbox(…);
ui.checkbox(…);
```

What is the user story behind this? Why would one want to change the text style of only some checkboxes?

PS: I think you linked to the wrong issue

**Comment by Mubelotix** [comment]
> What is the user story behind this?

I don't like how the widgets have larger font than body text. It does not integrate well. Especially since I changed the `TextStyle::Button` font size to a larger value because I needed to display larger items in a custom panel. I can't wait for #56 to be closed! 

> An alternative is to instead set a new default in Style. This would also make it more convenient to change it for a lot of checkboxes:

Your suggested solution would solve my problem. However, I wonder how complex a PR adding this would be. There are so many places where `text_style` is hardcoded. I would have to find and track all of them. Anyway I am already having problem with this in #407. That's why I switched back to a draft PR because as soon as I want to make it reviewable I discover another widget that lacks this feature. 

> Adding modifiers for them all may be the wrong approach.

Indeed. CSS implemented both solutions and I almost never use the element-specific one. However:
- Sometimes it's better to have it. Otherwise if we had like to change a single checkbox among others we would have to embed it into its own container. That's ugly.
- CSS is very powerful at selecting elements among others. It is not possible with `egui` yet (but that would be great). So the style-related solution would not be as great as in CSS. But you probably thought about this when you added other style fields.
